### PR TITLE
feat: 5701 dynamic value fetch for forms

### LINKED ISSUE #3951: Editor.json : Add functionality to read data into the form from datasource configuration or datasource structure

### LINKED ISSUE #5701: Implement functionality for dynamically triggering field values

### PR DESCRIPTION
**Pull request by ayushpahwa**
## Description
This change integrates the ability to read custom values into the forms. This functionality is majorly useful for the entity selector component and part of the Simplify crud queries project. This basically allows the options for a drop-down or any other value to be fetched from the server or any other end point based on a certain condition.

Fixes #5701
Fixes #3951

## Type of change

- New feature (non-breaking change which adds functionality)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: feat/5701-dynamic-value-fetch-for-forms 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 55.05 **(-0.05)** | 36.55 **(-0.03)** | 34.52 **(-0.02)** | 55.56 **(-0.04)**
 :red_circle: | app/client/src/api/PluginApi.ts | 85.71 **(-3.76)** | 100 **(0)** | 50 **(-10)** | 85 **(-3.89)**
 :green_circle: | app/client/src/components/formControls/DropDownControl.tsx | 44.83 **(0.39)** | 2.13 **(-0.04)** | 20 **(3.33)** | 42.86 **(0)**
 :red_circle: | app/client/src/pages/Editor/QueryEditor/EditorJSONtoForm.tsx | 35.94 **(-1.56)** | 28.9 **(-0.69)** | 0 **(0)** | 37.32 **(-1.68)**
 :red_circle: | app/client/src/sagas/FormEvaluationSaga.ts | 24.32 **(-15.68)** | 15 **(-12.27)** | 25 **(-25)** | 29.03 **(-18.03)**
 :red_circle: | app/client/src/workers/formEval.ts | 13.89 **(-0.4)** | 2.82 **(-0.31)** | 10 **(0)** | 13.89 **(-0.4)**</details>

### CODE REVIEW

File: app/client/src/pages/Editor/QueryEditor/EditorJSONtoForm.tsx [hunk 1]
```diff
@@ -645,15 +646,47 @@ export function EditorJSONtoForm(props: Props) {
...
 
+  // Function to modify the section config based on the output of evaluations
+  const modifySectionConfig = (
+    section: any,
+    enabled: boolean,
+    dynamicFetchedValues: DynamicValues | undefined,
+  ): any => {
+    let modifiedSection = { ...section };
+    if (!enabled) {
+      modifiedSection = { ...modifiedSection, disabled: true };
```
Comment (ApekshaBhosale) [inline app/client/src/pages/Editor/QueryEditor/EditorJSONtoForm.tsx:657]: why are you copying again and again?

File: app/client/src/workers/formEval.ts [hunk 2]
```diff
@@ -46,6 +49,10 @@ const generateInitialEvalState = (formConfig: FormConfig) => {
         ...conditionTypes,
         visible,
       };
+      conditionals = {
+        ...conditionals,
```
Comment (ApekshaBhosale) [inline app/client/src/workers/formEval.ts:53]: don't need to use separator you can just add to the object

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]

File: app/client/src/pages/Editor/QueryEditor/EditorJSONtoForm.tsx (hunk 1 above)
Comment (ayushpahwa) [reply app/client/src/pages/Editor/QueryEditor/EditorJSONtoForm.tsx]: Updated

File: app/client/src/workers/formEval.ts (hunk 2 above)
Comment (ayushpahwa) [reply app/client/src/workers/formEval.ts]: Updated

**Review by ApekshaBhosale — APPROVED** [review]

**Pull request merged**